visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -129,16 +137,48 @@ void MotionPlanningFrame::computePlanButtonClicked()
 void MotionPlanningFrame::computeExecuteButtonClicked()
 {
   if (move_group_ && current_plan_)
-    move_group_->execute(*current_plan_);
+  {
+    // ui_->stop_button->setEnabled(true); // enable stopping
+    bool success = move_group_->execute(*current_plan_);
+    ui_->result_label->setText(success ? "Executed" : "Execution failed");
+    ui_->stop_button->setEnabled(false); // disable stopping
+  }
+
+  // Hm. The current state is only updated after a while (only when using /execute/ service)
+  // TODO: Wait for update to planning scene? Related to #671
+  sleep(1);
+  updateStartStateToCurrent();
```

Comment:
If this does not explicitly ask move_group for the current state via a service request, :-1:.
Waiting for the current state via `sleep` is a workaround that might fail on slow systems/heavy load.
This is the exact problem I was trying to address by only sending empty start states in #710 .

In my opinion, a proper solution  would add a new "<always-current>" option and keep updating the maintained start state in this case. This has some non-trivial edge-cases, but it would allow for other nodes to move the arm without rendering the display's start-state obsolete.